awx/main/scheduler/task_manager.py

```diff
@@ -279,6 +279,8 @@ def create_project_update(self, task, project_id=None):
         # Project created 1 seconds behind
         project_task.created = task.created - timedelta(seconds=1)
         project_task.status = 'pending'
+        if task.scm_branch:
+            project_task.scm_branch = task.scm_branch
```

Comment:
Whether or not the part about setting the inventory update branch is a good idea (and it very well could be), I think applying the branch to the project update is not a good idea. If a job's `scm_branch` matches that of the project, then this is not necessary. If the job's `scm_branch` does not match the project, then it doesn't depend on the dependent project update in the first place. This would not do anything meaningful. We could, I think, avoid creating a project update dependency entirely in the case that `update_on_launch` is true, `branch_override` is true, and the job specifies a unique `scm_branch`. So in the case you're concerned about, ideally, we should never reach this code.

Reply:
I might have been overly careful here, or added this change before I tried the inventory update change. I can definitely check if removing this line of code still fixes things.